plugins/extractors/merlin/internal/merlin/merlin_client.go

```diff
@@ -149,15 +148,15 @@ func (c Client) exec(req *http.Request, result interface{}) error {
 
 func authenticatedClient(ctx context.Context, serviceAccountJSON []byte, scopes ...string) (*http.Client, error) {
 	if len(serviceAccountJSON) == 0 {
-		return google.DefaultClient(ctx, scopes...)
+		return http.DefaultClient, nil
 	}
 
-	creds, err := google.CredentialsFromJSON(ctx, serviceAccountJSON, authScopes...)
+	creds, err := idtoken.NewClient(ctx, authScope, idtoken.WithCredentialsJSON(serviceAccountJSON))
 	if err != nil {
 		return nil, fmt.Errorf("google credentials from JSON: %w", err)
 	}
 
-	return oauth2.NewClient(ctx, creds.TokenSource), nil
+	return creds, nil
```

Comment:
we should return the client instead of the creds
```suggestion
	return idtoken.NewClient(ctx, audience, idtoken.WithCredentialsJSON(serviceAccountJSON))
```

Reply:
creds is the Client mas, as of here: https://github.com/goto/meteor/pull/50/files/89ea7e069be23ee05127a436b35a798b12c13725#diff-bc097a9a691dad593c58dc5d74b40680cab5f94cdcfb41569d966f9d1a063bbfR154

Reply:
i see, let's keep the name consistent